api/end_device.proto

```diff
@@ -175,18 +173,79 @@ message EndDeviceVersion {
 }
 
 message MACSettings {
-  bool use_adr = 1 [(gogoproto.customname) = "UseADR"];
+  message DataRateIndexValue {
+    DataRateIndex value = 1;
+  }
+  message PingSlotPeriodValue {
+    PingSlotPeriod value = 1;
+  }
+  message AggregatedDutyCycleValue {
+    AggregatedDutyCycle value = 1;
+  }
+  message RxDelayValue {
+    RxDelay value = 1;
+  }
+
+  // Maximum delay for the device to answer a MAC request or a confirmed downlink frame.
+  // Must be set if the device supports class B.
+  google.protobuf.Duration class_b_timeout = 1 [(gogoproto.stdduration) = true];
+  // Periodicity of the class B ping slot.
+  // Must be set if the device supports class B.
+  PingSlotPeriodValue ping_slot_periodicity = 2;
+  // Data rate index of the class B ping slot.
+  // Must be set if the device supports class B.
+  DataRateIndexValue ping_slot_data_rate_index = 3;
+  // Frequency of the class B ping slot (Hz).
+  // Must be set if the device supports class B.
+  uint64 ping_slot_frequency = 4;
+
+  // Maximum delay for the device to answer a MAC request or a confirmed downlink frame.
+  // Must be set if the device supports class C.
+  google.protobuf.Duration class_c_timeout = 5 [(gogoproto.stdduration) = true];
+
+  // Class A Rx1 delay.
+  // Must be set if the device is ABP.
+  RxDelayValue rx1_delay = 6;
+  // Rx1 data rate offset.
+  // Must be set if the device is ABP.
+  google.protobuf.UInt32Value rx1_data_rate_offset = 7;
+  // Data rate index for Rx2.
+  // Must be set if the device is ABP.
+  DataRateIndexValue rx2_data_rate_index = 8;
+  // Frequency for Rx2 (Hz).
+  // Must be set if the device is ABP.
+  uint64 rx2_frequency = 9;
+  // List of factory-preset frequencies.
+  // Must be set if the device is ABP.
+  repeated uint64 factory_preset_frequencies = 10;
+
+  // Maximum uplink duty cycle (of all channels).
+  AggregatedDutyCycleValue max_duty_cycle = 11;
+
+  // Whether the device supports 32-bit frame counters.
+  // Must be set if the device MAC version is 1.0.
```

Comment:
@htdvisser @johanstokking do we need to enforce this? BE says we should, but it looks like we could also just use a NS default? I believe we don't break anything if we just always assume 32-bit counters?